tests/test_execution_compute_plan.py

```diff
@@ -150,46 +143,170 @@ def test_compute_plan_single_session_failure(global_execution_env):
     spec = factory.create_algo(py_script=sbt.factory.INVALID_ALGO_SCRIPT)
     algo = session.add_algo(spec)
 
-    cp_spec = factory.create_compute_plan(algo=algo, objective=objective)
+    cp_spec = factory.create_compute_plan(objective=objective)
 
     traintuple_spec_1 = cp_spec.add_traintuple(
+        algo=algo,
         dataset=dataset,
         data_samples=[data_sample_1]
     )
     cp_spec.add_testtuple(traintuple_spec_1)
 
     traintuple_spec_2 = cp_spec.add_traintuple(
+        algo=algo,
         dataset=dataset,
         data_samples=[data_sample_2],
-        traintuple_specs=[traintuple_spec_1]
+        in_models_tuples=[traintuple_spec_1]
     )
     cp_spec.add_testtuple(traintuple_spec_2)
 
     traintuple_spec_3 = cp_spec.add_traintuple(
+        algo=algo,
         dataset=dataset,
         data_samples=[data_sample_3],
-        traintuple_specs=[traintuple_spec_2]
+        in_models_tuples=[traintuple_spec_2]
     )
     cp_spec.add_testtuple(traintuple_spec_3)
 
     # Submit compute plan and wait for it to complete
-    cp = session.add_compute_plan(cp_spec)
+    cp_created = session.add_compute_plan(cp_spec)
+    cp = cp_created.future().wait()
```

Comment:
Could be done on a single line.
 We need also to add the option `raises=False` to ensure this function does not raise an exception as soon as the first tuple failed.

Reply:
I wanted to keep the cp as returned by the add_compute_plan function to make sure that the data returned by the add and get are the same ones (even if under different names). This is what the tests one lines 182-184 do.

Reply:
Yes, I haven't seen these lines. Sounds good to me.